src/transaction/log_recovery.c

```diff
@@ -1753,6 +1767,19 @@ log_rv_analysis_assigned_mvccid (THREAD_ENTRY * thread_p, int tran_id, LOG_LSA *
   auto rec = (const LOG_REC_ASSIGNED_MVCCID *) (log_page_p->area + log_lsa->offset);
   tdes->mvccinfo.id = rec->mvccid;
 
+  if (is_passive_transaction_server ())
+    {
+      /* Since there is no recovery redo phase on PTS, PTS does not update log_Gl.hdr.mvcc_next_id.
+       * So, largest_mvccid will be updated during log_recovery_analysis () only for PTS,
+       * and it will be used to set log_Gl.hdr.mvcc_next_id in log_recovery_analysis_from_trantable_snapshot ().
+       */
+
+      if (MVCC_ID_PRECEDES (context.get_largest_mvccid (), tdes->mvccinfo.id))
```

Comment:
Nit.
This comparison and set should be moved inside `set_largest_mvccid`.
Already the name `largest" suggests that there might be some condition verification inside. :)